python-checks/src/main/java/org/sonar/python/checks/hotspots/CsrfDisabledCheck.java

```diff
@@ -0,0 +1,326 @@
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.ASSIGNMENT_STMT, CsrfDisabledCheck::djangoMiddlewareArrayCheck);
+    context.registerSyntaxNodeConsumer(Tree.Kind.DECORATOR, CsrfDisabledCheck::decoratorCsrfExemptCheck);
+    context.registerSyntaxNodeConsumer(Tree.Kind.CALL_EXPR, CsrfDisabledCheck::functionCsrfExemptCheck);
+    context.registerSyntaxNodeConsumer(Tree.Kind.ASSIGNMENT_STMT, CsrfDisabledCheck::flaskWtfCsrfEnabledFalseCheck);
+    context.registerSyntaxNodeConsumer(Tree.Kind.CLASSDEF, CsrfDisabledCheck::metaCheck);
+    context.registerSyntaxNodeConsumer(Tree.Kind.CALL_EXPR, CsrfDisabledCheck::formInstantiationCheck);
+    context.registerSyntaxNodeConsumer(Tree.Kind.ASSIGNMENT_STMT, CsrfDisabledCheck::improperlyConfiguredFlaskApp);
+  }
+
+  private static final String CSRF_VIEW_MIDDLEWARE = "django.middleware.csrf.CsrfViewMiddleware";
+
+  /** Checks that <code>django.middleware.csrf.CsrfViewMiddleware</code> is in <code>MIDDLEWARE</code> array. */
+  private static void djangoMiddlewareArrayCheck(SubscriptionContext subscriptionContext) {
+    if (!"settings.py".equals(subscriptionContext.pythonFile().fileName())) {
+      return;
+    }
+
+    AssignmentStatement asgn = (AssignmentStatement) subscriptionContext.syntaxNode();
+    // Check that the left hand side is called `MIDDLEWARE` and that there is at least one string entry starting with
+    // "django" in that array.
+    boolean isLhsCalledMiddleware = isLhsCalled("MIDDLEWARE").test(asgn);
+    boolean containsDjangoMiddleware = isListAnyMatch(isStringSatisfying(s -> s.startsWith("django"))).test(asgn.assignedValue());
+    boolean isMiddlewareAssignment = isLhsCalledMiddleware && containsDjangoMiddleware;
+    if (isMiddlewareAssignment) {
+      boolean containsCsrfViewMiddleware = isListAnyMatch(isStringEqual(CSRF_VIEW_MIDDLEWARE)).test(asgn.assignedValue());
+      if (!containsCsrfViewMiddleware) {
+        subscriptionContext.addIssue(
+          asgn.lastToken(),
+          "Make sure not using CSRF protection (" + CSRF_VIEW_MIDDLEWARE + ") is safe here.");
+      }
+    }
+  }
+
+  /** Checks that the left hand side of the assignment is a variable with name <code>lhsName</code>. */
+  private static Predicate<AssignmentStatement> isLhsCalled(String lhsName) {
+    return asgn -> asgn.lhsExpressions().stream()
+      .flatMap(exprList -> exprList.expressions().stream())
+      .anyMatch(expr -> expr.is(Tree.Kind.NAME) && lhsName.equals(((Name) expr).name()));
+  }
+
+  /** Checks whether an expression is a string literal satisfying a predicate. */
+  private static Predicate<Expression> isStringSatisfying(Predicate<String> pred) {
+    return expr -> {
+      if (expr.is(Tree.Kind.STRING_LITERAL)) {
+        return pred.test(((StringLiteral) expr).trimmedQuotesValue());
+      } else {
+        return false;
```

Comment:
Minor nitpick but I believe the `else` is not needed since you return directly from the `if` anyway.